### ISSUE TITLE
Support withSchema [with PR]

### ISSUE THREAD

**Issue by Artaud** [feathersjs-ecosystem/feathers-sequelize#433]
I'd like to dynamically change schema based on request params, which can be done in Sequelize by calling the model with `withSchema`, similarly to `withScope` usage which is already supported by feathers-sequelize.

Is that something you would accept in feathers-sequelize?
I can propose a PR.

**Comment by Artaud** [comment]
At the currently supported version of sequelize, it would be just `schema` which is supposed to be deprecated by `withSchema` in sequelize 7

**Comment by Artaud** [comment]
I also see that you are currently getting to a new major release. I'd love to slip this addition in :)

**Comment by Artaud** [comment]
I'm not sure how to structure the addition though. I see that `applyScope` is deprecated in favor of `ModelWithScope`, which is being used as a model in all DB methods.
If I wanted to add `schema` support, I either can:

- add a new function similar to applyScope, like `applySchema`
- alter the `ModelWithScope` to be loosened to `ModelWithScopeAndSchema` which I really don't like for its smell, or 
- assume that support for more sequelize props can be added in the future, and make the Model function more generic such as:
```
ModelWithSequelizeParams(params: ServiceParams) {
  const Model = this.getModel(params)
  # apply scope if present in params
  # apply schema if present in params
  # here be space for future additions
  return Model
}
```
and then use this in all DB methods.

What do you think?

**Comment by DaddyWarbucks** [comment]
I am sorry for the slow response on this! I will investigate Sequelize's `withSchema` options and see if we can get that added.

**Comment by Artaud** [comment]
Thanks! I can see that you've got a lot on your hands (don't we all :D) so I completely understand, no rush :)
I've been already using the change i proposed in the PR in production (using patch-package) and it's been working good so far.

**Comment by DaddyWarbucks** [comment]
`ModelWithScope` is pretty ugly as a whole. We should just be using `getModel`. I need to fix a couple other little things and will add this.

**Comment by fratzinger** [comment]
I merged #450 and released as `v8.0.0-pre.2`.

@Artaud : Does this fit your use case? I don't use `withSchema`, `ModelWithScope` or `getModel` at all, so I cannot help with this.

**Comment by Artaud** [comment]
@fratzinger So this enables you to use any options that the underlying sequelize Model has or may have. That's a great idea, my use case is covered by that perfectly.

**Comment by fratzinger** [comment]
Are you already on the v8 pre release or which version of feathers-sequelize do you use?

**Comment by Artaud** [comment]
I'm on 7.0.2. Are you thinking about backporting it to 7?

**Comment by DaddyWarbucks** [comment]
I don't think we will be backporting this. My update to V8 was very easy and I am currently using it in an important production app.
